### PR TITLE
support double star arguments

### PR DESCRIPTION
**Pull request by uriva**

### CODE REVIEW

File: computation_graph/graph.py [hunk 1]
```diff
@@ -15,28 +16,48 @@ def _is_reducer_type(node: Callable) -> bool:
...
+            gamla.map(_parameter_name),
+            tuple,
         ),
     )
 
-    assert not any(
-        "**" + x.name == str(x)
-        for x in tuple(inspect.signature(func).parameters.values())
-    ), f"Cannot infer signature for {func}, signature includes **kwargs"
+    if gamla.anymap(_is_double_star)(function_parameters):
```
Comment (nitzo) [inline computation_graph/graph.py:59]: move this to the top of the function and then you can remove the `signature` param

File: computation_graph/graph.py [hunk 2]
```diff
@@ -70,37 +91,41 @@ def make_computation_node(
...
+    source: Union[_CallableOrNode, Tuple[_CallableOrNode, ...]],
+    destination: _CallableOrNode,
     key: Optional[Text] = None,
     priority: int = 0,
 ) -> base_types.ComputationEdge:
+    destination_as_node = make_computation_node(destination)
+    if destination_as_node.signature.is_kwargs:
+        assert not isinstance(source, tuple)
+        return base_types.ComputationEdge(
+            args=gamla.wrap_tuple(make_computation_node(source)),
```
Comment (nitzo) [inline computation_graph/graph.py:109]: why is this going to args ? why not in source ? can you explain ?

**Review by nitzo — APPROVED** [review]

File: computation_graph/graph.py (hunk 2 above)
Comment (uriva) [reply computation_graph/graph.py:109]: because we don't have a way to inject a nameless single variable. maybe you have a better idea?

Comment (nitzo) [reply computation_graph/graph.py:106]: in the args case we use `key=None`
you are saying if we set `source=source, destination=destination, key=None` it doesn't work ?

Comment (uriva) [reply computation_graph/graph.py:106]: iirc the behaviour on `None` is to infer the name of the variable from the signature, which in this case cannot be done

Comment (uriva) [reply computation_graph/graph.py]: changed the implementation, ptal

File: computation_graph/run.py [hunk 3]
```diff
@@ -201,6 +202,13 @@ def _get_computation_input(
 ) -> base_types.ComputationInput:
     bound_signature = _get_bound_signature(signature.is_args, incoming_edges)
     unbound_signature = _signature_difference(signature, bound_signature)
+    if signature.is_kwargs:
+        assert len(results) == 1
```
Comment (nitzo) [inline computation_graph/run.py:206]: maybe add an error message here ?

**Review by nitzo — APPROVED** [review]

File: computation_graph/run.py (hunk 3 above)
Comment (uriva) [reply computation_graph/run.py:206]: done

**Pull request merged**